app/models/vm_or_template/operations/relocation.rb

```diff
@@ -106,10 +106,27 @@ def raw_move_into_folder(folder)
     run_command_via_parent(:vm_move_into_folder, :folder => folder)
   end
 
-  def move_into_folder(folder)
+  def move_into_folder(folder_or_id)
+    folder = EmsFolder.find(folder_or_id) if folder_or_id.kind_of?(Integer)
```

Comment:
If a folder is passed in then `folder` will be nil.
Something like
```
folder = folder_or_id.kind_of?(Integer) ? EmsFolder.find(folder_or_id) : folder_or_id
```

Should do the trick